msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocationsTaskBase.cs

```diff
@@ -93,18 +93,21 @@ void EnsureSdkPath ()
 				Log.LogError ("Could not locate SDK bin directory");
 		}
 
-		void EnsureAppleSdkRoot ()
+		bool EnsureAppleSdkRoot ()
 		{
 			if (!MacOSXSdks.Native.IsInstalled) {
-				Log.LogError ("Could not find a valid Xcode app bundle");
-			} else {
-				Log.LogMessage (MessageImportance.Low, "DeveloperRoot: {0}", MacOSXSdks.Native.DeveloperRoot);
-				Log.LogMessage (MessageImportance.Low, "GetPlatformPath: {0}", MacOSXSdks.Native.GetPlatformPath ());
-
-				SdkDevPath = MacOSXSdks.Native.DeveloperRoot;
-				if (string.IsNullOrEmpty (SdkDevPath))
-					Log.LogError ("Could not find a valid Xcode developer path");
+				Log.LogError("Could not find a valid Xcode app bundle at '{0}'. Please update your Apple SDK location in Visual Studio's preferences (Project > SDK Locations > Apple > Apple SDK).", AppleSdkSettings.InvalidDeveloperRoot);
```

Comment:
This message will point to the wrong place if building from Windows, it should be generic or point to the Apple SDK setting depending on the IDE. For VS on Windows it's Tools > Options > Xamarin > iOS Settings > Apple SDK